runner/types.go

```diff
@@ -75,29 +75,41 @@ type Result struct {
 // function to get dsl variables from result struct
 func dslVariables() []string {
 	dslVarList := []string{}
+	sanitizeTag := func(tag string) {
+		if str := strings.Split(tag, ",")[0]; str != "" {
+			dslVarList = append(dslVarList, str)
+		}
+	}
 	t := reflect.TypeOf(Result{})
 	for i := 0; i < t.NumField(); i++ {
 		field := t.Field(i)
-		if field.Tag.Get("dsl") != "" {
-			dslVarList = append(dslVarList, strings.Split(field.Tag.Get("dsl"), ",")...)
+		if field.Tag.Get("json") != "" {
+			sanitizeTag(field.Tag.Get("json"))
 		}
-
-		if field.Name == "CSPData" {
-			u := reflect.TypeOf(httpx.CSPData{})
-			for j := 0; j < u.NumField(); j++ {
-				field := u.Field(j)
-				if field.Tag.Get("dsl") != "" {
-					dslVarList = append(dslVarList, strings.Split(field.Tag.Get("dsl"), ",")...)
+		if field.Type.Kind() == reflect.Ptr || field.Type.Kind() == reflect.Struct {
+			var types reflect.Type = field.Type
+			if field.Type.Kind() == reflect.Ptr {
+				types = field.Type.Elem()
+			}
+			if types.Kind() == reflect.Struct {
+				for j := 0; j < types.NumField(); j++ {
+					subField := types.Field(j)
+					if subField.Tag.Get("json") != "" {
+						sanitizeTag(subField.Tag.Get("json"))
+					}
 				}
 			}
 		}
+		if field.Name == "Hashes" {
+			dslVarList = append(dslVarList, "header_md5", "header_mmh3", "header_sha256", "header_simhash", "body_md5", "body_mmh3", "body_sha256", "body_simhash")
```

Comment:
It's not necessary to compare at each iteration, we can add them once before returning `dslVarList`

Reply:
I added this logic to preserve the position of md5, mmh3, sha256 and simhash right after hashes. 
Removed the comparison at each iteration.